notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,512 @@
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_noncausal,
+                                   model_type="MNL",
+                                   names=mnl_names_noncausal)
+
+num_vars_noncausal = len(reduce(lambda x, y: x + y, mnl_names_noncausal.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_noncausal.fit_mle(np.zeros(num_vars_noncausal),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_noncausal.get_statsmodels_summary()
+# -
+
+# ## Model 2
+
+# +
+mnl_specification_causal_1 = mnl_specification_noncausal.copy()
+mnl_names_causal_1 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_1["confounder_all"] = [1, 2, 3]
+mnl_names_causal_1["confounder_all"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_causal_1 = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_causal_1,
+                                   model_type="MNL",
+                                   names=mnl_names_causal_1)
+
+num_vars = len(reduce(lambda x, y: x + y, mnl_names_causal_1.values()))
+# Note newton-cg used to ensure convergence to a point where gradient 
+# is essentially zero for all dimensions. 
+mnl_model_causal_1.fit_mle(np.zeros(num_vars),
+                  method="BFGS")
+
+# Look at the estimation results
+mnl_model_causal_1.get_statsmodels_summary()
+# -
+
+# ## Model 3
+
+# +
+mnl_specification_causal_2 = mnl_specification_noncausal.copy()
+mnl_names_causal_2 = mnl_names_noncausal.copy()
+
+mnl_specification_causal_2["confounder_all_2"] = [1, 2, 3]
+mnl_names_causal_2["confounder_all_2"] = ["Confounder - Drive alone",
+                                     "Confounder - Shared ride 2", 
+                                     "Confounder - Shared ride 3"]
+
+# +
+# Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+mnl_model_causal_2 = cm.create_choice_model(data=data,
+                                   alt_id_col="mode_id",
+                                   obs_id_col="observation_id",
+                                   choice_col="sim_choice",
+                                   specification=mnl_specification_causal_2,
+                                   model_type="MNL",
+                                   names=mnl_names_causal_2)
```

Comment:
If there's no reason otherwise, I suggest following PEP8 [indentation guidelines](https://www.python.org/dev/peps/pep-0008/#indentation) by lining up all the arguments with `data=data`.